### ISSUE TITLE
Custom FPS cause FF Audio Only

### ISSUE THREAD

**Issue by hthetiot** [priologic/easyrtc#244]
easyrtc.setVideoDims with fps argument cause audio only on FF.

**Comment by MariasStory** [comment]
Hi @hthetiot,
I was not able to change fps in FF. Tried to report it and got finger pointing in different directions.

In my case the fps value has no effect on the video that I get. I also could not find an example that would getUsermedia in FF with different fps. Did you see such an example?

Would be cool if this option would finally work in FF.

Greetings,
Tolik

**Comment by EricDavies** [comment]
I've done some playing around with this today. The first thing to realize is that setVideoDims(width,height,framerate) is setting the maximum frame rate. Which is what we figured people probably wanted. Not much sense setting a minimum frame rate when the peer connection can (and usually does) throttle it down below that to meet network and processing limitations.

Second thing: If you set a really low maximum in Firefox, firefox simply fails to give you a video track. Set a high maximum (ie, something it can ignore) and it happily gives you a video track. In short, the problem is not that you can't set a custom frame rate, but that firefox won't give you a video track if you set constraints it can't meet. You can see the compliment of this (setting an overly high max) using the following demo: 
      https://webrtc.github.io/samples/src/content/peerconnection/constraints/

Yes, this is finger point at firefox, but it is finger pointing with reason and evidence :-).

**Comment by MariasStory** [comment]
Hi Eric,

Please check this one:
http://stackoverflow.com/questions/35516416/firefox-frame-rate-max-constraint

Setting max framerate lover than 30 in https://mikeg0.github.io/samples/src/content/peerconnection/constraints/
started only audio.

In fact, my use case is more of a practical. I would like to have 15 fps, because I see reasonable improvement in ISO of the camera and noise reduction. I am in Ubuntu and know that hardware supports the mode. Also it works perfectly in Chrome.

Greetings,
Tolik

**Comment by EricDavies** [comment]
Tolik, 
Unless I'm missing something those, two links just confirm what I said.

**Comment by MariasStory** [comment]
Hi Eric,

Agree. It might be that I misunderstood your point.

The problem is that Firefox isn't handling properly the max framerate property.

There seems to be some misconception on the FF side:
https://bugzilla.mozilla.org/show_bug.cgi?id=1286945

They argue that Chrome just grammatically lowers framerate form the camera.
I don't agree with this statement. The ISO of the camera is increased.

I thing that they should properly handle camera fps and not just decrease it programmatically. 

Greetings,
Tolik

**Comment by hthetiot** [comment]
@EricDavies Since update b604cb8dd5d23c804a194de4f1d627750814ca18#diff-db5890b7b83a9a5ba3f9bc3c6d33976cR709 I have issue getting video on Firefox 49.0.1

**Comment by EricDavies** [comment]
Harold, can you be more specific?
Are you have trouble getting video in general, or if you specify an unhandled frame rate?
I've just run the beta version on my machine, locally hosted, simple_audio_video_demo, and it had no trouble acquiring video.

**Comment by hthetiot** [comment]
@EricDavies No issue, sorry that was related to Firefox that lost the camera.

**Comment by hthetiot** [comment]
@EricDavies Here is the result of testing the `easyrtc.setVideoDims(640, 480, 25);` on FF 49.0.1.

Failed:
```
{"video":{"width":640,"height":480,"frameRate":{"ideal":25,"max":25}},"audio":{}}
```

Worked:
```
{"video":{"width":640,"height":480,"frameRate":{"minFrameRate":25,"maxFrameRate":25}},"audio":{}}
```

**Comment by hthetiot** [comment]
@EricDavies Test on you side my last commit an let me know.

**Comment by EricDavies** [comment]
@hthetiot I tried using  {"video":{"width":640,"height":480,"frameRate":{"ideal":25,"max":25}},"audio":true}. It worked fine. However, then I noticed that I was using firefox 44. So I pulled down firefox 49. It also worked fine. This is on linux (so things may be different on Windows or Mac).
Unless it's a behaviour that varies between operating systems, my guess is that your OS thought the camera was still in use producing some higher frame rate when you tried to ask it for 25fps, and it can only operate at one frame rate at a time. If you revert back to the "ideal"/"max" approach temporarily as a test, I suspect you'll find it to work. If that is the case, then the problem you observed earlier will likely reoccur at some point when you reproduce the same situation.
